openstudiocore/src/bimserver/ProjectImportation.cpp

```diff
@@ -0,0 +1,347 @@
+		QDialog setDialog(this);
+		QGridLayout *setLayout = new QGridLayout;
+		setDialog.setWindowTitle(tr("BIMserver Settings"));
+
+		QLabel *set_introLabel = new QLabel(tr("Please enter the BIMserver information: "), this);
+		QLabel *set_baddLabel = new QLabel(tr("BIMserver Address: http://"), this);
+		QLineEdit *set_baddEdit = new QLineEdit(this);
+		set_baddEdit->setPlaceholderText("eg: 127.0.0.1");
+		QLabel *set_bportLabel = new QLabel(tr("BIMserver Port:"), this);
+		QLineEdit *set_bportEdit = new QLineEdit(this);
+		set_bportEdit->setPlaceholderText("eg: 8082");
+		QLabel *set_unameLabel = new QLabel(tr("Username"), this);
+		QLineEdit *set_unameEdit = new QLineEdit(this);
+		set_unameEdit->setPlaceholderText("eg: admin@bimserver.org");
+		QLabel *set_upassLabel = new QLabel(tr("Password"), this);
+		QLineEdit *set_upassEdit = new QLineEdit(this);
+		set_upassEdit->setPlaceholderText("eg: admin");
+		QPushButton *set_okButton = new QPushButton(tr("Okay"), this);
+		QPushButton *set_cancelButton = new QPushButton(tr("Cancel"), this);
+
+		if (m_settings->contains("addr")) {
+			QString addr = m_settings->value("addr").toString();
+			set_baddEdit->setText(addr);
+		}
+
+		if (m_settings->contains("port")) {
+			QString port = m_settings->value("port").toString();
+			set_bportEdit->setText(port);
+		}
+
+		if (m_settings->contains("usrname")) {
+			QString usrname = m_settings->value("usrname").toString();
+			set_unameEdit->setText(usrname);
+		}
+
+		if (m_settings->contains("psw")) {
+			QString psw = m_settings->value("psw").toString();
+			set_upassEdit->setText(psw);
+		}
+
+		setLayout->addWidget(set_introLabel,0,0,1,2);
+		setLayout->addWidget(set_baddLabel,1,0,1,1);
+		setLayout->addWidget(set_baddEdit,1,1,1,1);
+		setLayout->addWidget(set_bportLabel,2,0,1,1);
+		setLayout->addWidget(set_bportEdit,2,1,1,1);
+		setLayout->addWidget(set_unameLabel,3,0,1,1);
+		setLayout->addWidget(set_unameEdit,3,1,1,1);
+		setLayout->addWidget(set_upassLabel,4,0,1,1);
+		setLayout->addWidget(set_upassEdit,4,1,1,1);
+		setLayout->addWidget(set_okButton,5,0,1,1);
+		setLayout->addWidget(set_cancelButton,5,1,1,1);
+
+		connect(set_okButton, SIGNAL(clicked()), &setDialog, SLOT(accept()));
+		connect(set_cancelButton, SIGNAL(clicked()), &setDialog, SLOT(reject()));
+
+		setDialog.setLayout(setLayout);
+
+		if (setDialog.exec()==QDialog::Accepted) {
+			QString address = set_baddEdit->text();
+			QString port = set_bportEdit->text();
+			QString usrname = set_unameEdit->text();
+			QString psw = set_upassEdit->text();
+
+			m_settings->setValue("addr", address);
+			m_settings->setValue("port", port);
+			m_settings->setValue("usrname",usrname);
+			m_settings->setValue("psw", psw);
+
+			if (!address.isEmpty() && !port.isEmpty() && !usrname.isEmpty() && !psw.isEmpty())
+			{ 
+				//in debug mode, this pointer is 0xCDCDCDCD, and is not null, creates an error
```

Comment:
fixed: I initialized m_bimserverConnection to be nullptr and give every other pointers a value in constructor.